lib/opentracing/span.rb

```diff
@@ -0,0 +1,59 @@
+module OpenTracing
+  # Span represents an OpenTracer Span
+  #
+  # See http://www.opentracing.io for more information.
+  class Span
+    NOOP_INSTANCE = Span.new.freeze
+    # Name of the operation
+    attr_accessor :operation_name
```

Comment:
It doesn't raze because the instance is frozen but its attributes aren't (I tried it). But you're right it should just be a no-op setter.